bpf/bpf_sock.c

```diff
@@ -982,11 +1013,20 @@ static __always_inline int __sock6_xlate_fwd(struct bpf_sock_addr *ctx,
 	if (sock6_skip_xlate(svc, &orig_key.address))
 		return -EPERM;
 
-	/* Do not perform service translation at socker layer for services with
-	 * L7 load balancing as the l7 load balancer needs to see the service IP.
-	 */
-	if (lb6_svc_is_l7loadbalancer(svc))
+#ifdef ENABLE_L7_LB
+	/* See __sock4_xlate_fwd for commentary. */
+	if (lb6_svc_is_l7loadbalancer(svc)) {
+		if (is_defined(BPF_HAVE_NETNS_COOKIE) && in_hostns) {
+			l7backend.address = { .addr[15] = 1, };
```

Comment:
Trying to set enable-envoy-config as true and check for IPv6 environment (e.g. kind conformance ipv6 test), but got the below error.

```
unknown-warning-option -Wno-gnu-variable-sized-type-not-at-end -Wdeclaration-after-statement -Wimplicit-int-conversion -Wenum-conversion -I. -I/run/cilium/state/globals -I/var/lib/cilium/bpf -I/var/lib/cilium/bpf/include -D__NR_CPUS__=32 -DENABLE_ARP_RESPONDER=1 -DCALLS_MAP=cilium_calls_lb -c /var/lib/cilium/bpf/bpf_sock.c -o -" subsys=datapath-loader
level=warning msg="+ llc -march=bpf -mcpu=v3 -mattr=dwarfris -filetype=obj -o bpf_sock.o" subsys=datapath-loader
level=warning msg="/var/lib/cilium/bpf/bpf_sock.c:1020:24: error: expected expression" subsys=datapath-loader
level=warning msg="                        l7backend.address = { .addr[15] = 1, };" subsys=datapath-loader
level=warning msg="                                            ^" subsys=datapath-loader

```

Reply:
@sayboras Thanks for the report, I pushed a new commit that fixes this.